packages/grafana-prometheus/src/datasource.ts

```diff
@@ -620,7 +620,7 @@ export class PrometheusDatasource
   // it is used in metric_find_query.ts
   // and in Tempo here grafana/public/app/plugins/datasource/tempo/QueryEditor/ServiceGraphSection.tsx
   async getTagKeys(options: DataSourceGetTagKeysOptions<PromQuery>): Promise<MetricFindValue[]> {
-    if (config.featureToggles.promQLScope && !!options) {
+    if (config.featureToggles.promQLScope && !!options && !!options.scopes && options.scopes.length > 0) {
```

Comment:
I'd simplify this a little bit

```suggestion
    if (config.featureToggles.promQLScope && (options?.scopes?.length ?? 0) > 0) {
```